pkg/storage/replica_proposal.go

```diff
@@ -413,7 +415,23 @@ func addSSTablePreApply(
 		// tell Rocks that it is not allowed to modify the file, in which case it
 		// will return and error if it would have tried to do so, at which point we
 		// can fall back to writing a new copy for Rocks to ingest.
-		if _, err := os.Stat(path); err == nil {
+		if stat, err := os.Stat(path); err == nil {
+			// HACK: RocksDB does not like ingesting the same file (by inode) twice.
+			// See facebook/rocksdb#5133. We can tell that we have tried to ingest
+			// this file already if it has more than one link – one from the file raft
+			// wrote and one from rocks. In that case, we should not try to give
+			// rocks a link to the same file again.
+			if sys := stat.Sys(); sys != nil {
+				if s, ok := sys.(*syscall.Stat_t); ok && s.Nlink == 1 {
```

Comment:
Does this compile on all platforms? The contents of the syscall package can vary and I'm not sure if it exists (and has an Nlink field) on windows. This may need to be in a build-tagged file.